### ISSUE TITLE
Add type related properties to MiscTable

### ISSUE THREAD

**Issue by hagenw** [audeering/audformat#300]
When looping through all tables of a database it might become easier if `MiscTable` would have a few more properties from `Table`.
For example, if you would like to gather some metadata in a single loop over all tables, and you would like to store how many filewise tables you have, you would need a check like:
```python
if not isinstance(table, audformat.MiscTable):
    if table.is_filewise:
        count += 1
```

First I thought it would be good to have
* `is_filewise`
* `is_segmented`
and set both to `False`, but I we could then have the following inconsitency:
```python
>>> table = audformat.MiscTable(pd.Index(["f1"], name="file", dtype="string"))
>>> table.is_filewise
False
>>> audformat.is_filewise_index(table)
True
```

So, I would maybe only thing about adding `type` and setting it to `'misc'`?

**Comment by frankenjoe** [comment]
If you loop over `db.tables` you don't have to check if the table is a misc table.

**Comment by hagenw** [comment]
> If you loop over `db.tables` you don't have to check if the table is a misc table.

Yes, but if you a lot of other stuff in the loop and some should affect tables and misc tables it becomes cumbersome to first loop tables and afterwards loop misc tables.

**Comment by frankenjoe** [comment]
> So, I would maybe only thing about adding type and setting it to 'misc'?

Ok, so the idea is that your example above can be simplified to:

```python
if not table.type == 'misc':
    if table.is_filewise:
        count += 1
```

?

**Comment by hagenw** [comment]
No, it would simplify to:

```python
if table.type == 'filewise':
    count += 1
```

**Comment by frankenjoe** [comment]
Ah wait, now I get it you want to introduce `'filewise'`, `'segmented'` and `'misc'`. I thought you only want to distinguish between `Table` and `MiscTable`, i.e. `'table'` and `'misc'` as `type`. Mhh, don't we otherwise run into the issue that `MiscTable` can still have a `filewise` index :)

**Comment by hagenw** [comment]
I changed the title of this issue to better track what we want to have.

It's indeed not completely straightforward as before we always derived the table type from the index, compare the current documentation of `Table.type`:

![image](https://user-images.githubusercontent.com/173624/213125823-bda48ec0-41f3-4afa-a048-fb040ad46305.png)

My proposal would be the following for `[MiscTable|Table].type`:

* for `Table` return `filewise` or `segmented`
* for `MiscTable` return `misc` independent of the used index

I guess that's more complicated from a consistency/documentation point of view, but would made writing code easier.

But I'm also not 100% sure we should add this cahnge.

**Comment by frankenjoe** [comment]
Yes, I am not sure if it is a good idea to mix up table type and index type in one field. That's why I intuitively expected that `type` would only have the values `table` and `misc`.  This would change your example to:

```python
if table.type == 'misc' and table.is_filewise:
    count += 1

**Comment by hagenw** [comment]
Yes, but `Table.type` is already in use and returns the index type (which we set identical to table type when the misc table was not existing).

So yes, maybe we don't do anything here and propose to use code like

```python
if isinstance(table, audformat.Table) and table.is_filewise:
    count += 1
```

**Comment by hagenw** [comment]
Then you also have the option if you want to count misc tables using a filewise index as well, e.g.

```python
if audformat.is_filewise_index(table.index):
    count += 1
```

**Comment by frankenjoe** [comment]
Agree
